<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="Docutils 0.8.1: http://docutils.sourceforge.net/" />
<title>Summary of ARMO shotgun read assembly:</title>
<meta name="author" content="Jiarong Guo" />
<meta name="date" content="2013-04-22" />
<style type="text/css">

/*
:Author: David Goodger (goodger@python.org)
:Id: $Id: html4css1.css 7056 2011-06-17 10:50:48Z milde $
:Copyright: This stylesheet has been placed in the public domain.

Default cascading style sheet for the HTML output of Docutils.

See http://docutils.sf.net/docs/howto/html-stylesheets.html for how to
customize this style sheet.
*/

/* used to remove borders from tables and images */
.borderless, table.borderless td, table.borderless th {
  border: 0 }

table.borderless td, table.borderless th {
  /* Override padding for "table.docutils td" with "! important".
     The right padding separates the table cells. */
  padding: 0 0.5em 0 0 ! important }

.first {
  /* Override more specific margin styles with "! important". */
  margin-top: 0 ! important }

.last, .with-subtitle {
  margin-bottom: 0 ! important }

.hidden {
  display: none }

a.toc-backref {
  text-decoration: none ;
  color: black }

blockquote.epigraph {
  margin: 2em 5em ; }

dl.docutils dd {
  margin-bottom: 0.5em }

object[type="image/svg+xml"], object[type="application/x-shockwave-flash"] {
  overflow: hidden;
}

/* Uncomment (and remove this text!) to get bold-faced definition list terms
dl.docutils dt {
  font-weight: bold }
*/

div.abstract {
  margin: 2em 5em }

div.abstract p.topic-title {
  font-weight: bold ;
  text-align: center }

div.admonition, div.attention, div.caution, div.danger, div.error,
div.hint, div.important, div.note, div.tip, div.warning {
  margin: 2em ;
  border: medium outset ;
  padding: 1em }

div.admonition p.admonition-title, div.hint p.admonition-title,
div.important p.admonition-title, div.note p.admonition-title,
div.tip p.admonition-title {
  font-weight: bold ;
  font-family: sans-serif }

div.attention p.admonition-title, div.caution p.admonition-title,
div.danger p.admonition-title, div.error p.admonition-title,
div.warning p.admonition-title {
  color: red ;
  font-weight: bold ;
  font-family: sans-serif }

/* Uncomment (and remove this text!) to get reduced vertical space in
   compound paragraphs.
div.compound .compound-first, div.compound .compound-middle {
  margin-bottom: 0.5em }

div.compound .compound-last, div.compound .compound-middle {
  margin-top: 0.5em }
*/

div.dedication {
  margin: 2em 5em ;
  text-align: center ;
  font-style: italic }

div.dedication p.topic-title {
  font-weight: bold ;
  font-style: normal }

div.figure {
  margin-left: 2em ;
  margin-right: 2em }

div.footer, div.header {
  clear: both;
  font-size: smaller }

div.line-block {
  display: block ;
  margin-top: 1em ;
  margin-bottom: 1em }

div.line-block div.line-block {
  margin-top: 0 ;
  margin-bottom: 0 ;
  margin-left: 1.5em }

div.sidebar {
  margin: 0 0 0.5em 1em ;
  border: medium outset ;
  padding: 1em ;
  background-color: #ffffee ;
  width: 40% ;
  float: right ;
  clear: right }

div.sidebar p.rubric {
  font-family: sans-serif ;
  font-size: medium }

div.system-messages {
  margin: 5em }

div.system-messages h1 {
  color: red }

div.system-message {
  border: medium outset ;
  padding: 1em }

div.system-message p.system-message-title {
  color: red ;
  font-weight: bold }

div.topic {
  margin: 2em }

h1.section-subtitle, h2.section-subtitle, h3.section-subtitle,
h4.section-subtitle, h5.section-subtitle, h6.section-subtitle {
  margin-top: 0.4em }

h1.title {
  text-align: center }

h2.subtitle {
  text-align: center }

hr.docutils {
  width: 75% }

img.align-left, .figure.align-left, object.align-left {
  clear: left ;
  float: left ;
  margin-right: 1em }

img.align-right, .figure.align-right, object.align-right {
  clear: right ;
  float: right ;
  margin-left: 1em }

img.align-center, .figure.align-center, object.align-center {
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.align-left {
  text-align: left }

.align-center {
  clear: both ;
  text-align: center }

.align-right {
  text-align: right }

/* reset inner alignment in figures */
div.align-right {
  text-align: inherit }

/* div.align-center * { */
/*   text-align: left } */

ol.simple, ul.simple {
  margin-bottom: 1em }

ol.arabic {
  list-style: decimal }

ol.loweralpha {
  list-style: lower-alpha }

ol.upperalpha {
  list-style: upper-alpha }

ol.lowerroman {
  list-style: lower-roman }

ol.upperroman {
  list-style: upper-roman }

p.attribution {
  text-align: right ;
  margin-left: 50% }

p.caption {
  font-style: italic }

p.credits {
  font-style: italic ;
  font-size: smaller }

p.label {
  white-space: nowrap }

p.rubric {
  font-weight: bold ;
  font-size: larger ;
  color: maroon ;
  text-align: center }

p.sidebar-title {
  font-family: sans-serif ;
  font-weight: bold ;
  font-size: larger }

p.sidebar-subtitle {
  font-family: sans-serif ;
  font-weight: bold }

p.topic-title {
  font-weight: bold }

pre.address {
  margin-bottom: 0 ;
  margin-top: 0 ;
  font: inherit }

pre.literal-block, pre.doctest-block, pre.math {
  margin-left: 2em ;
  margin-right: 2em }

span.classifier {
  font-family: sans-serif ;
  font-style: oblique }

span.classifier-delimiter {
  font-family: sans-serif ;
  font-weight: bold }

span.interpreted {
  font-family: sans-serif }

span.option {
  white-space: nowrap }

span.pre {
  white-space: pre }

span.problematic {
  color: red }

span.section-subtitle {
  /* font-size relative to parent (h1..h6 element) */
  font-size: 80% }

table.citation {
  border-left: solid 1px gray;
  margin-left: 1px }

table.docinfo {
  margin: 2em 4em }

table.docutils {
  margin-top: 0.5em ;
  margin-bottom: 0.5em }

table.footnote {
  border-left: solid 1px black;
  margin-left: 1px }

table.docutils td, table.docutils th,
table.docinfo td, table.docinfo th {
  padding-left: 0.5em ;
  padding-right: 0.5em ;
  vertical-align: top }

table.docutils th.field-name, table.docinfo th.docinfo-name {
  font-weight: bold ;
  text-align: left ;
  white-space: nowrap ;
  padding-left: 0 }

h1 tt.docutils, h2 tt.docutils, h3 tt.docutils,
h4 tt.docutils, h5 tt.docutils, h6 tt.docutils {
  font-size: 100% }

ul.auto-toc {
  list-style-type: none }

</style>
</head>
<body>
<div class="document" id="summary-of-armo-shotgun-read-assembly">
<h1 class="title">Summary of ARMO shotgun read assembly:</h1>
<table class="docinfo" frame="void" rules="none">
<col class="docinfo-name" />
<col class="docinfo-content" />
<tbody valign="top">
<tr><th class="docinfo-name">Author:</th>
<td>Jiarong Guo</td></tr>
<tr class="field"><th class="docinfo-name">Tags:</th><td class="field-body">bioinformatics, assembly, metagenomics</td>
</tr>
<tr><th class="docinfo-name">Date:</th>
<td>2013-04-22</td></tr>
<tr class="field"><th class="docinfo-name">Slug:</th><td class="field-body">asseSum</td>
</tr>
<tr class="field"><th class="docinfo-name">Category:</th><td class="field-body">science</td>
</tr>
</tbody>
</table>
<p>(with Adina Howe, James Tiedje, Titus Brown)</p>
<p>I have been working on the assembly of big shotgun metagenomic data from ARMO (Amazon Rain Forest Microbial  Observatory) project. The biggest challenge is the huge data size, 2TB in fastq and more than 6 billions reads after read trimming. One lucky thing is that MSU provides High Performance Computer Clusters, so we have access to memory up to 1TB. Still, to my knowledge, no assemblers can handle this amount of data with 1Tb of memory. Well, some tools like SGA may be able to handle the data if they are very redundant, but this is not the case for soil metagenomic data. <a class="reference external" href="http://arxiv.org/abs/1203.4802">Digital normalization</a>
and <a class="reference external" href="http://arxiv.org/abs/1112.4193">partitioning</a>, two data preprocessing methods  designed by our lab (<a class="reference external" href="http://ged.msu.edu/">Dr. Titus Brown</a>) are used to make the assembly doable.</p>
<div class="section" id="background-and-data">
<h1>Background and data</h1>
<p>Amazon rain forest is very important for global ecosystem and microbial communities play key roles in the nutrient cycling. The goal of this study is  to understand the effect of agriculture practice (conversion of forest to pasture) on microbial community. A survey based on 16s rRNA gene have shown that the land conversion cause higher local microbial diversity but more similar community across space. To further study the microbial community at functional level, shotgun metagenomics is used.</p>
<p>To evaluate the effects of spatial variation, samples are taken at corner of nested 0.01m, 0.1m, 1m, 10m, 100m square plot at forest site and pasture site. Each sample from a square plot are sequenced by two Illumina HighSeq lanes.</p>
<div class="figure">
<img alt="figs/expDesign.png" src="figs/expDesign.png" style="width: 571.0px; height: 311.0px;" />
<p class="caption">Figure 1. Experiment design. Samples are collected at corners are nested squares of 0.01, 0.1, 1, 10, 100 meters. Details are in this <a class="reference external" href="http://www.pnas.org/content/110/3/988/F2.expansion.html">paper</a></p>
</div>
</div>
<div class="section" id="methods">
<h1>Methods</h1>
<p>Data trimming: the end of reads with quality score 2 (ASCII “B”) are trimmed and reads with length longer than 30bp are kept. Pair end reads with overlap are stitched with FLASH (version 1.0.3).</p>
<p>Digital normalization, sequencing artifact removing, and partitioning: This step is done by a bash script (<a class="reference external" href="https://github.com/jiarong/khmer/blob/master/ged-lab/armo-gjr/asse6.bash">https://github.com/jiarong/khmer/blob/master/ged-lab/armo-gjr/asse6.bash</a>).</p>
<p>Assembly: VELVET assembler is used to assemble the group of sequence file with kmer size of  33, 37, 39, 49, 69. The one producing the most overall assembly length is picked. Another version of VELVET compiled with “BIGASSEMBLY” flag is used for assembly the biggest group of sequence file.</p>
</div>
<div class="section" id="results">
<h1>Results</h1>
<div class="section" id="computational-cost">
<h2>Computational cost</h2>
<p>Digital normalization: 150 cpu hours to finish 10 lanes of Illumina HighSeq data for each treatment (forest or pasture) with 1TB memory.</p>
<p>Artifact removal: 40 cpu hours to finish 10 lanes.</p>
<p>Partitioning: 1) 69 cpu hours for loading graph; 2) 346 cpu hours for partitioning graph (12 hours with 32 core on HPC); 3) 12 cpu hours for merging graph; 4) 42 cpu hours for annotating partition; 5) 13 cpu hours for extracting partitions. In total, the partitioning took one week to finish, because the most time consuming step, partitioning graph can be run with multi-threads.</p>
<p>Assembly: There is always one group file that is significantly larger than the others after partitioning because reads from different species could be connected by some conserved genes or artifacts. We call it the “lump”. The “lump” took about 20 cpu hours, while the other small group files took less than 0.5 cpu hours.</p>
<div class="figure">
<img alt="figs/flowChart.png" src="figs/flowChart.png" style="width: 519.6px; height: 190.8px;" />
<p class="caption">Figure 2: Flow chart of the assembling process.</p>
</div>
</div>
<div class="section" id="genomic-saturation">
<h2>Genomic saturation</h2>
<p>Figure 3, 4 shows there is a fair amount of shared sequence among samples within the same treatment (forest), which also suggests the data within the treatment can be combined for better assemblies.</p>
<div class="figure">
<img alt="figs/FdigiNormByLane.report.png" src="figs/FdigiNormByLane.report.png" style="width: 480.0px; height: 360.0px;" />
<p class="caption">Figure 3. Genomic content saturation curve from digital normalization output from forest samples. The X axis the number of reads processed by digital normalization. The Y axis the fraction of reads kept after digital normalization in the current sample data.</p>
</div>
<div class="figure">
<img alt="figs/PdigiNormByLane.report.png" src="figs/PdigiNormByLane.report.png" style="width: 480.0px; height: 360.0px;" />
<p class="caption">Figure 4. Genomic content saturation curve from digital normalization output from the pasture samples. The X axis the number of reads processed by digital normalization. The Y axis the fraction of reads kept after digital normalization in the current sample data.</p>
</div>
</div>
<div class="section" id="assembly-statistics">
<h2>Assembly statistics</h2>
<table border="1" class="docutils">
<caption>Table 1: Assembly statistics. Minimum contig length of 800 bp is chosen. The low mapping percentage is due to the high minimum contig length cutoff.</caption>
<colgroup>
<col width="11%" />
<col width="22%" />
<col width="14%" />
<col width="8%" />
<col width="13%" />
<col width="14%" />
<col width="17%" />
</colgroup>
<thead valign="bottom">
<tr><th class="head">Sample</th>
<th class="head">&gt;800bp contigs</th>
<th class="head">total bp</th>
<th class="head">max</th>
<th class="head">mapping</th>
<th class="head">mapping%</th>
<th class="head">total reads</th>
</tr>
</thead>
<tbody valign="top">
<tr><td>Forest</td>
<td>265073</td>
<td>271553890</td>
<td>9115</td>
<td>10732078</td>
<td>0.37%</td>
<td>2923068636</td>
</tr>
<tr><td>Pasture</td>
<td>497664</td>
<td>538669724</td>
<td>24861</td>
<td>23082813</td>
<td>0.81%</td>
<td>2863547487</td>
</tr>
</tbody>
</table>
<p>As shown in Table 1, pasture data have more and longer contigs assembled (with 800bp cutoff). The DNA content in the two metagenome assemblies share little similarity (Table 2). Rank abundance curves (Figure 5) show the pasture assembly has better coverage than forest assembly.</p>
<table border="1" class="docutils">
<caption>Table 2 Similarity between two assemblies. Contigs covered is the fraction of total contigs covered by any contigs from the other sample. Total bp covered is the fraction of total basepairs covered by contigs from the other sample.</caption>
<colgroup>
<col width="11%" />
<col width="23%" />
<col width="15%" />
<col width="25%" />
<col width="26%" />
</colgroup>
<thead valign="bottom">
<tr><th class="head">Sample</th>
<th class="head">&gt;800bp contigs</th>
<th class="head">total bp</th>
<th class="head">contigs covered</th>
<th class="head">total bp covered</th>
</tr>
</thead>
<tbody valign="top">
<tr><td>Forest</td>
<td>265073</td>
<td>271553890</td>
<td>9.30%</td>
<td>3.50%</td>
</tr>
<tr><td>Pasture</td>
<td>497664</td>
<td>538669724</td>
<td>5.20%</td>
<td>1.80%</td>
</tr>
</tbody>
</table>
<div class="figure">
<img alt="figs/FvsP_randAbun.png" src="figs/FvsP_randAbun.png" style="width: 480.0px; height: 360.0px;" />
<p class="caption">Figure 5. Rank abundance of forest and pasture samples. Bule curve ( F.800.contigCov) is from the combined forest assembly with 800 minimum length and green curve (P.800.contigCov) is from the combined pasture assembly with 800 minimum length. The coverage is based on median of coverages on each base position in contigs after mapping.</p>
</div>
</div>
<div class="section" id="spatial-variation">
<h2>Spatial variation</h2>
<p>Clustering and heatmap of samples (Figure 6, 7) in each treatment based on the fold coverage of top 1000 most abundant contig shows that even though sharing a fair amount of genomic content, there are still spatial  variation between samples. The closer the distance, the more similar the samples are.</p>
<div class="figure">
<img alt="figs/F.1500bp.cov.table.dendrogram.hellinger.png" src="figs/F.1500bp.cov.table.dendrogram.hellinger.png" style="width: 480.0px; height: 480.0px;" />
<p class="caption">Figure 6. Heatmap of top 1000 most abundant contigs within forest samples. Samples are in the rows. “F” stands for forest. The number follow the “F” stands for the distance of sampling spot. For example, “F001” is 0.01m at forest. Contigs are the columns (name not shown due to large amount). The columns are sorted based on the abundance in F01 for A. The abundance matrix is transformed by hellinger transform prior to clustering.</p>
</div>
<div class="figure">
<img alt="figs/P.1500bp.cov.table.dendrogram.hellinger.png" src="figs/P.1500bp.cov.table.dendrogram.hellinger.png" style="width: 480.0px; height: 480.0px;" />
<p class="caption">Figure 7. Heatmap of top 1000 most abundant contigs within pasture samples. Samples are in the rows. “P” stands for pasture. The number follow the “P” stands for the distance of sampling spot. For example, “P001” is 0.01m at pasture. Contigs are the columns (name not shown due to large amount). The columns are sorted based on the abundance in P01 for pasture. The abundance matrix is transformed by hellinger transform prior to clustering.</p>
</div>
</div>
</div>
<div class="section" id="conclusion">
<h1>Conclusion</h1>
<p>Digital normalization and partitioning are effective methods to assemble large metagenomic data. The assemblies from combined forest samples and from combined pasture samples share less than 5% similarity, which indicates the metagenomic content in forest and pasture are quite different. The microbial communities across space share a fair amount of genomic content, but they are still distinct from each other, especially when distance are large. The assembly data has been uploaded to MG-RAST for annotation. Next step will be comparing the communities with gene or functional category rather than just the contigs.</p>
</div>
</div>
</body>
</html>
